edge-functions/feature-flag-posthog/middleware.ts

```diff
@@ -0,0 +1,70 @@
+// eslint-disable-next-line @next/next/no-server-import-in-page
+import { NextRequest, NextResponse } from 'next/server'
+import { DISTINCT_ID_COOKIE_NAME, FEATURE_FLAGS } from '@lib/constants'
+import { getFeatureFlagVariant, isFeatureFlagEnabled } from '@lib/posthog-node'
+
+export const config = {
+  matcher: ['/product', '/about', '/marketing'],
+}
+
+export default async function middleware(req: NextRequest) {
+  const url = req.nextUrl
+
+  if (url.pathname.startsWith('/product')) {
+    // Defaults to product
+    url.pathname = '/product'
+    if (url.pathname != '/product') {
+      // Redirect paths that go directly to the variant
+      return NextResponse.redirect(url)
+    }
```

Comment:
In this case and in the ones similar below, we no longer have to check for the variant or use `startsWith`, the matcher makes sure we're only going to get requests for `/product`